lite2/client.go

```diff
@@ -727,23 +727,25 @@ func (c *Client) sequence(
 		}
 
 		// Update trusted header and vals.
+		trustedHeader = interimHeader
```

Comment:
we need to set them both together. otherwise, if valset is invalid, trustedHeader will be new, trustedNextValSet - old